src/main/kotlin/domain/usecase/GetIngredientGuessUseCase.kt

```diff
@@ -0,0 +1,50 @@
+package org.example.domain.usecase
+
+import org.example.domain.repository.FoodRepository
+import org.example.model.Food
+import org.example.model.IngredientGameRound
+
+class GetIngredientGuessUseCase  (private val repository : FoodRepository) {
+    private fun getAllFood(): List<Food> = repository.getFood()
+
+    private var score = 0
+    private var correctAnswers = 0
+    private val usedMeals = mutableSetOf<String>()
+    private var currentRound: IngredientGameRound? = null
+
+    fun nextRound(): IngredientGameRound?  {
+            val meal = getRandomMeal() ?:return null
+            usedMeals.add(meal.name!!)
```

Comment:
this could cause  NullPointerException what about using a let to safely add non-null name
```suggestion
            meal.name?.let { usedMeals.add(it) }
```

Reply:
The previous line ensures that if the meal is null, it will return null and won’t reach the line usedMeals.add(meal.name!!). That’s why I’m confident the name won’t be null, because I’ve already validated it earlier.